knowhere unittest need check CPU instruction set firstly

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
git action report error as following:
```
test_annoy  running...
/milvus/ci/scripts/run_unittest.sh: line 100: 16444 Illegal instruction     (core dumped) ${UNITTEST_DIR}/${test}
Error: Process completed with exit code 132.
```

**Steps/Code to reproduce behavior**
 git action sometimes report this error

**Expected behavior**
no error

--- a/core/src/index/unittest/utils.h
+++ b/core/src/index/unittest/utils.h
@@ -19,8 +19,15 @@
 
 #include "knowhere/common/Dataset.h"
 #include "knowhere/common/Log.h"
+#include "faiss/FaissHook.h"
 
 class DataGen {
+ public:
+    DataGen() {
+        std::string cpu_flag;
+        faiss::hook_init(cpu_flag);
+    }
+
  protected:
     void
     Init_with_default(const bool is_binary = false);

